SqlAssignments.sql

```diff
@@ -0,0 +1,1846 @@
+
+
+
+Que 35 -> Write a select command using correlated subquery that selects the names and numbers of all customers with ratings equal to the maximumfor their city.
+
+
+Query ->SELECT cnum,cname,city,rating 
+      ->FROM Customers 
+      ->WHERE rating IN(SELECT MAX(rating)                
+                        FROM Customers                
+                        GROUP BY city);
++------+----------+---------+--------+
+| cnum | cname    | city    | rating |
++------+----------+---------+--------+
+| 2001 | hoffman  | london  |    100 |
+| 2002 | Giovanni | Rome    |    200 |
+| 2003 | Liu      | SanJose |    200 |
+| 2004 | Grass    | Berlin  |    300 |
+| 2006 | Clemens  | London  |    100 |
+| 2007 | Pereira  | Rome    |    100 |
+| 2008 | Cisneros | SanJose |    300 |
++------+----------+---------+--------+
+7 rows in set (0.00 sec)
+
+
+
+
+
+**************************************************************************************************************************************
+
+
+
+Que 36 -> Write a query that totals the orders for each day and places the results in descending order.
+
+Query  -> SELECT SUM(amt) AS total_amt,odate 
+       -> FROM Orders 
+       -> GROUP BY odate 
+       -> ORDER BY odate DESC;
++-----------+------------+
+| total_amt | odate      |
++-----------+------------+
+|  11201.83 | 1990-06-10 |
+|      4723 | 1990-05-10 |
+|   1788.98 | 1990-04-10 |
+|   8944.59 | 1990-03-10 |
++-----------+------------+
+4 rows in set (0.00 sec)
+ 
+
+
+**************************************************************************************************************************************
+
+Que 37 -> Write a select command that produces the rating followed by the name of each customer in SanJose.
+
+Query -> SELECT cnum,cname,rating 
+      -> FROM Customers 
+      -> WHERE city ='SanJose';
++------+----------+--------+
+| cnum | cname    | rating |
++------+----------+--------+
+| 2003 | Liu      |    200 |
+| 2008 | Cisneros |    300 |
++------+----------+--------+
+2 rows in set (0.00 sec)
+
+
+**************************************************************************************************************************************
+
+Que 38 -> Find all orders with above average amounts for their customers.
+
+Query  -> SELECT onum,amt,cnum
+       -> FROM Orders 
+       -> WHERE amt > (SELECT AVG(amt)
+       ->              FROM Orders);
++------+---------+------+
+| onum | amt     | cnum |
++------+---------+------+
+| 3005 | 5160.45 | 2003 |
+| 3008 |    4723 | 2006 |
+| 3011 | 9891.88 | 2006 |
++------+---------+------+
+3 rows in set (0.00 sec)
+
+
+
+**************************************************************************************************************************************
+
+Que 39 ->Find all orders with amounts smaller than any amount for a customer in SanJose. 
+
+Query  -> SELECT onum,amt,odate,cnum 
```

Comment:
Why the condition of cname='Liu' has been used in this query.

Reply:
I should use any keyword in this query.